habitat-baselines/habitat_baselines/config/default_structured_configs.py

```diff
@@ -0,0 +1,366 @@
+    scheduled_std: bool = False
+
+
+@dataclass
+class ObsTransformConfig(HabitatBaselinesBaseConfig):
+    pass
+
+
+@dataclass
+class CenterCropperConfig(ObsTransformConfig):
+    height: int = 256
+    width: int = 256
+    channels_last: bool = True
+    trans_keys: Tuple[str] = (
+        "rgb",
+        "depth",
+        "semantic",
+    )
+
+
+@dataclass
+class ResizeShortestEdgeConfig(ObsTransformConfig):
+    size: int = 256
+    channels_last: bool = True
+    trans_keys: Tuple[str] = (
+        "rgb",
+        "depth",
+        "semantic",
+    )
+    semantic_key: str = "semantic"
+
+
+@dataclass
+class Cube2EqConfig(ObsTransformConfig):
+    height: int = 256
+    width: int = 512
+    sensor_uuids: List[str] = field(default_factory=lambda: [
+        "BACK",
+        "DOWN",
+        "FRONT",
+        "LEFT",
+        "RIGHT",
+        "UP",
+    ])
+
+
+@dataclass
+class Cube2FishConfig(ObsTransformConfig):
+    height: int = 256
+    width: int = 256
+    fov: int = 180
+    params: Tuple[float] = (0.2, 0.2, 0.2)
+    sensor_uuids: List[str] = field(default_factory=lambda: [
+        "BACK",
+        "DOWN",
+        "FRONT",
+        "LEFT",
+        "RIGHT",
+        "UP",
+    ])
+
+
+@dataclass
+class Eq2CubeConfig(ObsTransformConfig):
+    height: int = 256
+    width: int = 256
+    sensor_uuids: List[str] = field(default_factory=lambda: [
+        "BACK",
+        "DOWN",
+        "FRONT",
+        "LEFT",
+        "RIGHT",
+        "UP",
+    ])
+
+
+@dataclass
+class PolicyConfig(HabitatBaselinesBaseConfig):
+    name: str = "PointNavResNetPolicy"
+    action_distribution_type: str = "categorical"  # or 'gaussian'
+    # If the list is empty, all keys will be included.
+    # For gaussian action distribution:
+    action_dist: ActionDistributionConfig = ActionDistributionConfig()
+    obs_transforms: Any = MISSING
+
+
+@dataclass
+class PPOConfig(HabitatBaselinesBaseConfig):
+    """Proximal policy optimization config"""
+    clip_param: float = 0.2
+    ppo_epoch: int = 4
+    num_mini_batch: int = 2
+    value_loss_coef: float = 0.5
+    entropy_coef: float = 0.01
+    lr: float = 2.5e-4
+    eps: float = 1e-5
+    max_grad_norm: float = 0.5
+    num_steps: int = 5
+    use_gae: bool = True
+    use_linear_lr_decay: bool = False
+    use_linear_clip_decay: bool = False
+    gamma: float = 0.99
+    tau: float = 0.95
+    reward_window_size: int = 50
+    use_normalized_advantage: bool = False
+    hidden_size: int = 512
+    entropy_target_factor: float = 0.0
+    use_adaptive_entropy_pen: bool = False
+    use_clipped_value_loss: bool = True
+    # Use double buffered sampling, typically helps
+    # when environment time is similar or larger than
+    # policy inference time during rollout generation
+    # Not that this does not change the memory requirements
+    use_double_buffered_sampler: bool = False
+
+
+@dataclass
+class VERConfig(HabitatBaselinesBaseConfig):
```

Comment:
Should VER inherit from PPOConfig ? Where does the VERConfig get its learning rate for example?

Reply:
In the `default.py` it doesn't. See:

https://github.com/facebookresearch/habitat-lab/blob/112e50a6c59e022d665377e4c0d23ec392a99872/habitat-baselines/habitat_baselines/config/default.py#L187

https://github.com/facebookresearch/habitat-lab/blob/112e50a6c59e022d665377e4c0d23ec392a99872/habitat-baselines/habitat_baselines/config/default.py#L216

Reply:
`VERTrainer` does inherit from `PPOTrainer`, but accepts the top-level config and uses `ppo_cfg = self.config.RL.PPO` and `self.ver_config = self.config.RL.VER`.